rules/opentitan_test.bzl

```diff
@@ -297,24 +300,43 @@ def opentitan_functest(
 
     deps = kwargs.pop("deps", [])
     all_tests = []
+    target_params = {}
 
     # Only build SW for devices we are running tests on.
     devices_to_build_for = []
     for target in targets:
         if target == "dv":
             devices_to_build_for.append("sim_dv")
+            target_params["sim_dv"] = dv_params() if not dv else dv
         elif target == "verilator":
             devices_to_build_for.append("sim_verilator")
-        elif target == "cw310":
+            target_params["sim_verilator"] = verilator_params() if not verilator else verilator
+        elif target in ["cw310_rom", "cw310_test_rom"]:
             devices_to_build_for.append("fpga_cw310")
+
+            # Deep-copy the `cw310` dict.
+            cw310_ = cw310_params() if not cw310 else {k: v for k, v in cw310.items()}
```

Comment:
if we are going to split the cw310 device into two devices (cw310_{rom,test_rom}) does it make sense to also get rid of the "rom_kind" parameter in the cw310_params() generated dict? it seems redundant now no?

Reply:
I am not sure, since the target `cw310_rom` or `cw310_test_rom` eventually gets translated into a `cw310`, which expects both the bitstream and rom_kind parameters.

That being said, this whole file needs to be cleaned up soon :)